The shipped doctest module has an incorrect module test.

This patch fixes the module for cases with decorated methods.
--- Python-2.5.1-orig/Lib/doctest.py	2008-05-23 21:41:22.000000000 +1200
+++ Python-2.5.1/Lib/doctest.py	2008-05-23 21:42:00.000000000 +1200
@@ -835,12 +835,12 @@
         """
         if module is None:
             return True
+        elif inspect.getmodule(object) is not None:
+            return module is inspect.getmodule(object)
         elif inspect.isfunction(object):
             return module.__dict__ is object.func_globals
         elif inspect.isclass(object):
             return module.__name__ == object.__module__
-        elif inspect.getmodule(object) is not None:
-            return module is inspect.getmodule(object)
         elif hasattr(object, '__module__'):
             return module.__name__ == object.__module__
         elif isinstance(object, property):
